### ISSUE TITLE
Add documentation for zone.js in Angular

### ISSUE THREAD

**Issue by jasonkenneth** [angular/angular#28493]
# 📚 Docs or angular.io bug report

### Description

I'm new to angular but familiar with rxjs. I was caught by surprise when I sent a value through an observable (subject) via the console, but the change was not reflected on the screen. At the end of a lot of searching and learning, I ran across this link that fixed the issue for me whereby I could enable zone patches in the angular polyfill.ts file: [https://stackoverflow.com/questions/50784775/purpose-of-zone-js-dist-zone-patch-rxjs](https://stackoverflow.com/questions/50784775/purpose-of-zone-js-dist-zone-patch-rxjs).

It would be nice if the docs mentioned something to the effect of `vanilla rxjs is being used, the UI change detection is being managed by zones.js, and if you'd like your rxjs pipelines to be zone-aware, enable this patch in the polyfill.ts`. Or if there are any side-effects of doing so, document what those are.

Or maybe include them in the polyfill.ts project template:

```typescript
/***************************************************************************************************
 * Zone JS is required by Angular itself.
 */
import 'zone.js/dist/zone';  // Included with Angular CLI.

// uncomment below to make sure that rx streams are zone-aware and change detection gets properly triggered
//import 'zone.js/dist/zone-patch-rxjs';
//import 'zone.js/dist/zone-patch-rxjs-fake-async';
```

### What's the affected URL?

https://stackblitz.com/edit/angular-jasonkenneth-iybrkf?file=src%2Fapp%2Fasync-pipe%2Fasync-pipe.component.ts

**Comment by JiaLiPassion** [comment]
@jasonkenneth, thanks for posting the issue, the `zone-rxjs patching` has some bugs, I will add the document after I fix the bug in `zone.js` first.

**Comment by jenniferfell** [comment]
Thanks. We might need to mention this in both browser support (where we discuss polyfills) and the observables docs.

**Comment by JiaLiPassion** [comment]
@jenniferfell, got it, will do.

**Comment by kapunahelewong** [comment]
Hi, @JiaLiPassion, is this related to #30939?

**Comment by JiaLiPassion** [comment]
@kapunahelewong, no, this `zone-patch-rxjs` related to several pending issues in `angular/zone.js` repo, such as 
1. https://github.com/angular/zone.js/issues/1018
2. https://github.com/angular/zone.js/issues/1227

I need to fix those first to make this `patch` stable and add the doc later. 
And I will check the use case @jenniferfell provided to understand the scenario better.

thanks!

**Comment by kapunahelewong** [comment]
Ah, thank you for the clarification @JiaLiPassion. When you are ready, I'm happy to help. Please ping me then and we can collaborate. 🌼

**Comment by jelbourn** [comment]
I'm updating this issue to track adding documentation for zone.js overall rather than one specific part of zones.

**Comment by JeanMeche** [comment]
With Zoneless becoming stable in v20.2 and soon becoming the default, I don't see us expending documentation on zone.js.
